src/common/models/time-shift/time-shift.ts

```diff
@@ -83,4 +84,22 @@ export class TimeShift implements Instance<TimeShiftValue, TimeShiftJS> {
   toString(): string {
     return this.toJS() || "";
   }
+
+  private isValidForTimeFilter(timeFilter: TimeFilterClause, timezone: Timezone): boolean {
+    switch (timeFilter.type) {
+      case FilterTypes.FIXED_TIME:
+        const { values } = timeFilter as FixedTimeFilterClause;
+        const range = values.first();
+        return !range.intersects(range.shift(this.value, timezone));
+      case FilterTypes.RELATIVE_TIME:
+        const { duration } = timeFilter as RelativeTimeFilterClause;
+        return this.value.getCanonicalLength() >= duration.getCanonicalLength();
+      default:
+        throw new Error(`Incorrect time filter: ${timeFilter}`);
+    }
+  }
+
+  constrainToFilter(timeFilter: TimeFilterClause, timezone: Timezone): TimeShift {
```

Comment:
Does it make sense to make `Timezone.UTC` a default value here? It'd make tests a bit more readable.

Reply:
No. I don't think changing API for test convenience is a good idea.